h2o-py/tests/testdir_algos/gbm/pyunit_4007_decision_tree_language_rules_printing.py

```diff
@@ -0,0 +1,84 @@
+    print(first_tree.language_tree_representation)
+    text_file = open("pyunit_4007_language_tree_representation_numerical_case.txt", "r")
+    expected_tree_representation = text_file.read()
+    text_file.close()
+    assert expected_tree_representation == first_tree.language_tree_representation
+    
+    print(" -- Language path representation - root node: -- ")
+    assert first_tree.language_path_representations[first_tree.root_node.id] is not None
+    print(first_tree.language_path_representations[first_tree.root_node.id])
+    
+    print(" -- Language path representation - node ", first_tree.predictions.index(first_tree.predictions[3]), " (with pv = ", first_tree.predictions[3], "): -- ")
+    assert first_tree.language_path_representations[first_tree.predictions.index(first_tree.predictions[3])] is not None
+    print(first_tree.language_path_representations[first_tree.predictions.index(first_tree.predictions[3])])
+    text_file = open("pyunit_4007_language_path_representation_numerical_case.txt", "r")
+    expected_tree_representation = text_file.read()
+    text_file.close()
+    assert expected_tree_representation == first_tree.language_path_representations[first_tree.predictions.index(first_tree.predictions[3])]
+    
+    print(" ----- categorical case: -----")
+    airlines_data = h2o.import_file(path=pyunit_utils.locate("smalldata/airlines/allyears2k_headers.zip"))
+    model = H2OGradientBoostingEstimator(ntrees = 3, max_depth = 2, seed = 12345)
+    model.train(x=["Origin", "Distance"], y="IsDepDelayed", training_frame=airlines_data)
+    tree = H2OTree(model = model, tree_number = 0, tree_class = "NO")
+
+    print(" -- Tree predictions: -- ")
+    print(tree.predictions)
+    
+    print(" -- Language tree representation: -- ")
+    assert tree.language_tree_representation is not None
+    print(tree.language_tree_representation)
+    text_file = open("pyunit_4007_language_tree_representation_categorical_case.txt", "r")
+    expected_tree_representation = text_file.read()
+    text_file.close()
+    assert expected_tree_representation == tree.language_tree_representation
+
+    print(" -- Language path representation - root node: -- ")
+    assert tree.language_path_representations[tree.root_node.id] is not None
+    print(tree.language_path_representations[tree.root_node.id])
+    
+    print(" -- Language path representation - node ", tree.predictions.index(tree.predictions[3]), " (with pv = ", tree.predictions[3], "): -- ")
+    assert tree.language_path_representations[tree.predictions.index(tree.predictions[3])] is not None
+    print(tree.language_path_representations[tree.predictions.index(tree.predictions[3])])
+    text_file = open("pyunit_4007_language_path_representation_categorical_case.txt", "r")
+    expected_tree_representation = text_file.read()
+    text_file.close()
+    assert expected_tree_representation == tree.language_path_representations[tree.predictions.index(tree.predictions[3])]
```

Comment:
repeats a bunch of times what about assert read_fixture("file_name.txt") == tree.language_path...

Reply:
fixed